### PR TITLE
V2 omnipay mollie

### PR DESCRIPTION
**Pull request by ibudasov**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,13 +29,19 @@
...
+    },
     "require": {
         "omnipay/common": "^3"
     },
     "require-dev": {
         "omnipay/tests": "^3",
         "squizlabs/php_codesniffer": "^3",
-        "phpro/grumphp": "^0.14"
+        "phpro/grumphp": "^0.14",
+        "phpmd/phpmd": "@stable",
```
Comment (barryvdh) [inline composer.json]: Can you provide versions for this? Or is there a specific reason to not require a version?

File: composer.json [hunk 2]
```diff
@@ -29,13 +29,19 @@
...
         "psr-4": { "Omnipay\\Mollie\\" : "src/" }
     },
+    "autoload-dev": {
+        "psr-4": { "Omnipay\\Mollie\\Test\\": "tests/" }
+    },
     "require": {
         "omnipay/common": "^3"
     },
     "require-dev": {
         "omnipay/tests": "^3",
```
Comment (barryvdh) [inline composer.json]: Can you bump this to ^3.1, that should fix the lowest dependencies. (I've updated the php-mock-client dependency in 3.1)

File: composer.json [hunk 3]
```diff
@@ -29,13 +29,19 @@
...
         "squizlabs/php_codesniffer": "^3",
-        "phpro/grumphp": "^0.14"
+        "phpro/grumphp": "^0.14",
+        "phpmd/phpmd": "@stable",
+        "overtrue/phplint": "@stable",
+        "jakub-onderka/php-parallel-lint": "@stable"
     },
     "extra": {
         "branch-alias": {
```
Comment (barryvdh) [inline composer.json]: Can you bump the branch-alias to 5.0.x-dev? Because of breaking changes etc.

File: src/Gateway.php [hunk 4]
```diff
@@ -48,82 +69,82 @@ public function setApiKey($value)
 
     /**
      * @param  array $parameters
-     * @return \Omnipay\Mollie\Message\FetchIssuersRequest
+     * @return AbstractRequest|FetchIssuersRequest
```
Comment (barryvdh) [inline src/Gateway.php]: Isn't it redundant to add `AbstractRequest` as the `FetchIssuersRequest` already extends that? (Same for the others below)

**Comment by barryvdh** [comment]
Thanks! Looks great, just a few nitpicks and it's good to merge :)
A bit too bad that Github doesn't understand all moved files so it's a bit hard to see the actual difference in files.

**Comment by barryvdh** [comment]
Sorry, I meant `omnipay/tests:^3.1`, common was fine already with just `^3`

File: composer.json [hunk 5]
```diff
@@ -33,19 +33,19 @@
...
     "require-dev": {
         "omnipay/tests": "^3",
         "squizlabs/php_codesniffer": "^3",
         "phpro/grumphp": "^0.14",
-        "phpmd/phpmd": "@stable",
-        "overtrue/phplint": "@stable",
-        "jakub-onderka/php-parallel-lint": "@stable"
+        "phpmd/phpmd": "^2",
+        "overtrue/phplint": "^1",
+        "jakub-onderka/php-parallel-lint": "^0"
```
Comment (barryvdh) [inline composer.json]: There is a 1.x release out: https://packagist.org/packages/jakub-onderka/php-parallel-lint

File: composer.json (hunk 2 above)
Comment (barryvdh) [reply composer.json]: (referring to omnipay/tests)

**Comment by ibudasov** [comment]
@barryvdh could you please have a look at the pipeline? 
What can I do better?

**Comment by barryvdh** [comment]
Not really sure why he is complaining about not being able to install 3.0.1
You could try enforcing omnipay/common 3.0.2, not sure if that fixes it though.

**Comment by ibudasov** [comment]
Yay! It works!

**Comment by barryvdh** [comment]
Cool, so this is ready to merge?

**Comment by willemstuursma** [comment]
@roelvanduijnhoven FYI

**Comment by barryvdh** [comment]
This is tagged as 5.0.0 now

**Comment by roelvanduijnhoven** [comment]
@willemstuursma Thanks :). This helps a lot!

**Comment by willemstuursma** [comment]
Np @roelvanduijnhoven. Contact me at willem@mollie.com if you need technical support and I'll get you in touch with the right people.

**Pull request merged**